### DISCUSSION TITLE
Two-phase class initialization (allow code before super calls)

### DISCUSSION THREAD

**Discussion by peq** [wurstscript/WurstScript#1018] [Ideas]
Phase1: Not possible to use 'this' (no method calls etc.)
super-call
Phase2: Everything allowed

Also look at [Declaring and Checking Non-null Types in an Object-Oriented Language
Manuel Fähndrich and K. Rustan M. Leino](http://research.microsoft.com/en-us/um/people/leino/papers/krml109.pdf), section 3 and section 9

**Comment by Frotty** [comment]
yes yes yes
